unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapZoomViewController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Cysharp.Threading.Tasks;
+using DCLServices.MapRendererV2.MapCameraController;
+using System;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace DCL
+{
+    public class NavmapZoomViewController : IDisposable
+    {
+        private const float MOUSE_WHEEL_THRESHOLD = 0.04f;
+
+        private readonly NavmapZoom view;
+
+        private bool active;
+
+        private CancellationTokenSource cts;
+        private bool isScaling;
+
+        private float targetNormalizedZoom;
+        private int currentZoomLevel;
+
+        private IMapCameraController cameraController;
+
+        private readonly AnimationCurve normalizedCurve;
+        private readonly int zoomSteps;
+
+        public NavmapZoomViewController(NavmapZoom view)
+        {
+            this.view = view;
+
+            // Keys should be between [0;1]
+            var keys = view.normalizedZoomCurve.keys;
+
+            var firstKey = keys[0];
+            firstKey.time = 0;
+            firstKey.value = 0;
+            keys[0] = firstKey;
+
+            var lastKey = keys[^1];
+            lastKey.value = 1;
+            keys[^1] = lastKey;
+
+            normalizedCurve = new AnimationCurve(keys);
+            zoomSteps = keys.Length;
+        }
+
+        public float ResetZoom()
+        {
+            SetZoomLevel(Mathf.FloorToInt((zoomSteps - 1) / 2f));
+            return targetNormalizedZoom;
+        }
+
+        public void Activate(IMapCameraController mapCameraController)
```

Comment:
We have to handle multiple calls to `Activate` (without deactivation) to avoid leaking CTS instances.

Reply:
Same about multiple subscription to MouseWheelAction/ZoomIn/ZoomOut events.